jbpm-human-task/jbpm-human-task-workitems/src/main/java/org/jbpm/services/task/wih/AbstractHTWorkItemHandler.java

```diff
@@ -224,12 +224,13 @@ protected ContentData createTaskContentBasedOnWorkItemParams(KieSession session,
         return data;
     }
     
-    protected boolean isAutoClaim(WorkItem workItem, Task task) {
-        String swimlaneUser = (String) workItem.getParameter("SwimlaneActorId");
-        if (swimlaneUser != null  && !"".equals(swimlaneUser) && task.getTaskData().getStatus() == Status.Ready) {
-            return true;
+    protected boolean isAutoClaim(KieSession session, WorkItem workItem, Task task) {
+        if(Boolean.parseBoolean((String)session.getEnvironment().get("Autoclaim"))) {
```

Comment:
This is changing default behavior where autoclaim should be active by default. This should remain the same way. So I’d say autonclaim should not be used when it is explicitly disabled via environment entry

Reply:
Done